### PR TITLE
cannon: update cannon location when interacting with the player's cannon

### LINKED ISSUE #8106: Cannon Overlay not working at Cave Horrors
Please check if your issue is not a duplicate by [searching existing issues](https://github.com/runelite/runelite/search?type=Issues)

**Describe the bug**
The cannon overlay does not work in the mos le harmless caves, in particular when cannoning cave horrors.

**To Reproduce**
1. Go to Mos Le Harmless caves.
(not sure if it makes a difference but I am not wearing a witchwood icon)
2. Place cannon in the recommended cannon spot.

**Expected behavior**
The cannon will be setup but there will be no overlay like there is in other places. The cannonball indicator tile does not update either.

**Screenshots**
image

**Environment (please complete the following information):**
 - OS: Windows 10
 - RuneLite version: 1.5.14
 - Launcher version: 1.6.0

**Additional context**
Add any other context about the problem here, e.g. logs. Your client logs can usually be found in
`C:\Users\<your_user_name>\.runelite\logs` on Windows and `~/.runelite/logs` on Linux and macOS.

### PR DESCRIPTION
**Pull request by Hydrox6**
When interacting with another player's cannon, a completely different
message is returned. Using this, if we ever get a normal cannon message
after clicking on a cannon, we can be 100% certain that it is ours.

Closes #8106 by allowing the player to fix it without a re-place

Also cleaned up the mess that is `onChatMessage`, which was doing an unnecessary amount of string operations on cannon-related messages.

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/cannon/CannonPlugin.java [hunk 1]
```diff
@@ -257,6 +262,19 @@ public void onGameObjectSpawned(GameObjectSpawned event)
 			}
 		}
 	}
+	
+	@Subscribe
+	public void onMenuOptionClicked(MenuOptionClicked event)
+	{
+		// Check for the Fire option being selected on the cannon.
+		if (event.getMenuAction() != MenuAction.GAME_OBJECT_FIRST_OPTION || event.getId() != ObjectID.DWARF_MULTICANNON)
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/cannon/CannonPlugin.java:270]: This should also support using cannonballs on a cannon as that would also reload a cannon.

**Review by Nightfirecat — CHANGES_REQUESTED** [review]

File: runelite-client/src/main/java/net/runelite/client/plugins/cannon/CannonPlugin.java (hunk 1 above)
Comment (Hydrox6) [reply runelite-client/src/main/java/net/runelite/client/plugins/cannon/CannonPlugin.java]: Good shout. Fixed, and also found (and fixed) a weird tick-perfect issue where the indicator could appear on another cannon.

File: runelite-client/src/main/java/net/runelite/client/plugins/cannon/CannonPlugin.java [hunk 2]
```diff
@@ -77,6 +81,8 @@
 	private CannonCounter counter;
 	private boolean skipProjectileCheckThisTick;
 	private boolean cannonBallNotificationSent;
+	private WorldPoint clickedCannonLocation;
+	private boolean firstCannonLoad = false;
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/cannon/CannonPlugin.java:85]: `false` is the default initialized value for fields.
```suggestion
	private boolean firstCannonLoad;
```

**Review by Nightfirecat — COMMENTED** [review]
LGTM

**Pull request merged**